alerting/alerter-war/src/main/java/org/hawkular/metrics/alerter/ConditionEvaluator.java

```diff
@@ -0,0 +1,312 @@
+     * @return true if expression resolves to true, otherwise false
+     */
+    public boolean evaluate() {
+        return (this.expression.eval() != BigDecimal.ZERO);
+    }
+
+    /**
+     * Just a convenience method to prepare and evaluate in one call.
+     * @return true if expression resolves to true, otherwise false
+     */
+    public boolean prepareAndEvaluate(Map<String, BucketPoint> queryMap) {
+        prepare(queryMap);
+        return (this.expression.eval() != BigDecimal.ZERO);
+    }
+
+    private double getValue(String func, BucketPoint bucketPoint) throws Exception {
+        if (bucketPoint instanceof AvailabilityBucketPoint) {
+            return getAvailabilityValue(func, (AvailabilityBucketPoint) bucketPoint);
+        }
+
+        return getNumericValue(func, (NumericBucketPoint) bucketPoint);
+    }
+
+    private double getNumericValue(String func, NumericBucketPoint data) {
+        if (null == data || null == data.getSamples() || 0 == data.getSamples()) {
+            throw new IllegalArgumentException("NumericBucketPoint has no samples");
+        }
+
+        if (func.startsWith("%")) {
+            return funcToPercentile(func, data.getPercentiles()).getValue();
+        }
+
+        try {
+            switch (Function.valueOf(func)) {
+                case avg:
+                    return data.getAvg().doubleValue();
+                case max:
+                    return data.getMax().doubleValue();
+                case median:
+                    return data.getMedian().doubleValue();
+                case min:
+                    return data.getMin().doubleValue();
+                case samples:
+                    return data.getSamples() * 1.0;
+                case sum:
+                    return data.getSum().doubleValue();
+                default:
+                    throw new IllegalArgumentException("Unexpected func [" + func + "]");
+            }
+        } catch (IllegalArgumentException e) {
+            throw e;
+        } catch (Exception e) {
+            throw new IllegalArgumentException("Invalid value for func [" + func + "] in " + data.toString());
+        }
+    }
+
+    private double getAvailabilityValue(String func, AvailabilityBucketPoint data) {
+        if (null == data || null == data.getSamples() || 0 == data.getSamples()) {
+            throw new IllegalArgumentException("AvailabilityBucketPoint has no samples");
+        }
+
+        try {
+            switch (Function.valueOf(func)) {
+                case notUpCount:
+                    return data.getNotUpCount().doubleValue();
+                case notUpDuration:
+                    return data.getNotUpDuration().doubleValue();
+                case upCount:
+                    return data.getUpCount().doubleValue();
+                case uptimeRatio:
+                    return data.getUptimeRatio().doubleValue();
+                case samples:
+                    return data.getSamples() * 1.0;
```

Comment:
Multiplying by a double literal always returns a double, so a cast is not necessary.